src/Sylius/Bundle/ShopBundle/Resources/config/routing/partial/product.yml

```diff
@@ -1,13 +1,28 @@
 # This file is part of the Sylius package.
 # (c) Paweł Jędrzejewski
 
-sylius_shop_partial_product_index:
-    path: /
+sylius_shop_partial_product_index_latest:
+    path: /latest/{count}
     methods: [GET]
     defaults:
         _controller: sylius.controller.product:indexAction
         _sylius:
+            template: $template
+            repository:
+                method: findLatestByChannel
+                arguments:
+                    - "expr:service('sylius.context.channel').getChannel()"
+                    - $count
```

Comment:
I wanted it to be explicit, otherwise the whole array is a bit too much yaml magic. :D